pandas/core/window.py

```diff
@@ -883,104 +946,15 @@ class _Rolling(_Window):
     def _constructor(self):
         return Rolling
 
-    def _apply(
-        self, func, name=None, window=None, center=None, check_minp=None, **kwargs
-    ):
-        """
-        Rolling statistical measure using supplied function.
-
-        Designed to be used with passed-in Cython array-based functions.
-
-        Parameters
-        ----------
-        func : str/callable to apply
-        name : str, optional
-           name of this function
-        window : int/array, default to _get_window()
-        center : bool, default to self.center
-        check_minp : function, default to _use_window
-
-        Returns
-        -------
-        y : type of input
-        """
-        if center is None:
-            center = self.center
-        if window is None:
-            window = self._get_window()
-
-        if check_minp is None:
-            check_minp = _use_window
-
-        blocks, obj, index = self._create_blocks()
-        block_list = list(blocks)
-        index, indexi = self._get_index(index=index)
-
-        results = []
-        exclude = []
-        for i, b in enumerate(blocks):
-            try:
-                values = self._prep_values(b.values)
-
-            except (TypeError, NotImplementedError):
-                if isinstance(obj, ABCDataFrame):
-                    exclude.extend(b.columns)
-                    del block_list[i]
-                    continue
-                else:
-                    raise DataError("No numeric types to aggregate")
-
-            if values.size == 0:
-                results.append(values.copy())
-                continue
-
-            # if we have a string function name, wrap it
-            if isinstance(func, str):
-                cfunc = getattr(libwindow, func, None)
-                if cfunc is None:
-                    raise ValueError(
-                        "we do not support this function "
-                        "in libwindow.{func}".format(func=func)
-                    )
-
-                def func(arg, window, min_periods=None, closed=None):
-                    minp = check_minp(min_periods, window)
-                    # ensure we are only rolling on floats
-                    arg = ensure_float64(arg)
-                    return cfunc(arg, window, minp, indexi, closed, **kwargs)
-
-            # calculation function
-            if center:
-                offset = _offset(window, center)
-                additional_nans = np.array([np.NaN] * offset)
-
-                def calc(x):
-                    return func(
-                        np.concatenate((x, additional_nans)),
-                        window,
-                        min_periods=self.min_periods,
-                        closed=self.closed,
-                    )
+    def _apply_roll(self, func, values):
 
+        with np.errstate(all="ignore"):
```

Comment:
could this of be the same for Window?

Reply:
apply_along_axis is used only for Window. Else part fails not sure why

Reply:
reason was cython function returning memory view. solved